python/ray/tests/test_autoscaler.py

```diff
@@ -204,6 +204,9 @@ def is_terminated(self, node_id):
             return self.mock_nodes[node_id].state in ["stopped", "terminated"]
 
     def node_tags(self, node_id):
+        # Don't assume that node providers can retrieve tags from terminated nodes.
+        if self.is_terminated(node_id):
+            raise Exception(f"The node with id {node_id} has been terminated!")
```

Comment:
Need to watch to see if this causes test flakiness -- I noticed that node updaters in `test_autoscaler.py` sometimes fail as a result of this addition (but all of the tests pass anyway when I run them locally).

Reply:
Lets make this a warning then?

Reply:
Would rather keep as is -- it validates the bug fix.

This behavior is meant to catch failures of the kind that Allen saw -- specifically, autoscaler trying to read node tags of non-existent nodes.

I think it would be better to fix any test failures that arise from this, if we see them.

Worst case, we could make raising an exception here configurable and enable it in the relevant tests.